core/src/test/java/io/questdb/griffin/AsOfJoinTest.java

```diff
@@ -266,7 +266,7 @@ public void testAsOfJoinNoAliasDuplication() throws Exception {
                             "FROM (select b.bid b, b.ts timebid from bids b) b \n" +
                             "    ASOF JOIN\n" +
                             "(select a.ask a, a.ts timeask from asks a) a\n" +
-                            "    ON (b.timebid != a.timeask);";
+                            "WHERE (b.timebid != a.timeask);";
```

Comment:
why do we need to change tests ? is `ON` no-longer valid syntax ?

Reply:
ON works but not for arbitrary expressions . 
Such conditions were treated as inner join and pushed elsewhere , e.g. post-join or parent where clause which wasn't correct because asof is a left join .
Implementing asof with arbitrary expressions would require many more changes (e.g. scanning slave cursors backwards or resetting it for each master item) so this piece only makes the limitations more visible .

Reply:
is expression "arbitrary" because it is `!=` rather than `=` ?

Reply:
In this context - yes but join conditions could be anything .

Reply:
To make it more clear - this change actually preserves query outcome  because in master branch most join conditions except equality are pushed to post-join or parent where clause (regardless of join type) .

Reply:
This test passes with and without WHERE. The condition simply does not filer anything so even with the semantics changed the result is the same.

Having it in WHERE makes sense since we error that don't support having not equals in ON.

Reply:
@bziobrowski any comments ?

Reply:
Not really . Condition doesn't affect test output but if we change it then output is consistent with current master . 
I discussed the thing with Alex on slack .